core/src/main/java/com/linecorp/armeria/server/auth/AuthService.java

```diff
@@ -84,8 +90,18 @@ public static AuthServiceBuilder builder() {
         this.defaultFailureHandler = defaultFailureHandler;
     }
 
+    @Override
+    public void serviceAdded(ServiceConfig cfg) throws Exception {
+        super.serviceAdded(cfg);
+        final MeterRegistry meterRegistry = cfg.server().meterRegistry();
+        timer = Timer.builder("armeria.server.auth")
```

Comment:
If there are multiple `AuthService`s loaded in a single JVM, I think this timer may not produce accurate results.

What do you think of providing a way for users to specify the meter name with a `MeterIdPrefixFunction` in `AuthServiceBuilder`?

Reply:
For more context, my modifications look like:

My AuthServiceBuilder:
```
public AuthServiceBuilder meterIdPrefixFunction(MeterIdPrefixFunction meterIdPrefixFunction) {
    this.meterIdPrefixFunction = requireNonNull(meterIdPrefixFunction, "meterIdPrefixFunction");
    return this;
}
```

My AuthService:
```
@Override
public void serviceAdded(ServiceConfig cfg) throws Exception {
    super.serviceAdded(cfg);
    final MeterRegistry meterRegistry = cfg.server().meterRegistry();

    final String meterName = meterIdPrefixFunction.activeRequestPrefix(meterRegistry, null).name(); # null for requestOnlyLog
    timer = Timer.builder(meterName).register(meterRegistry);
}
```

Reply:
@jrhee17, Isn't `MeterIdPrefix` enough for this case?
```java
public AuthServiceBuilder meterIdPrefix(MeterIdPrefix meterIdPrefix) { ...}


class AuthService ...
    ...
    private final MeterIdPrefix meterIdPrefix;
    ...
    AuthService(HttpService delegate, Authorizer<HttpRequest> authorizer, MeterIdPrefix meterIdPrefix, ...) {...}

    @Override
    public HttpResponse serve(ServiceRequestContext ctx, HttpRequest req) throws Exception {
        final long startNanos = System.nanoTime();
        return HttpResponse.of(AuthorizerUtil.authorizeAndSupplyHandlers(authorizer, ctx, req)
                                             .handleAsync((result, cause) -> {
            try {
                ...
            } finally {
                MoreMeters.newTimer(ctx.meterRegistry(), meterIdPrefix.name(), meterIdPrefix.tags())
                          .record(System.nanoTime() - startNanos, TimeUnit.NANOSECONDS);
            }
        }, ctx.eventLoop()));
    }
    ...
}    
```

Reply:
I realized that `MeterIdPrefixFunction` would provide more information so we need it.
```java

class AuthService ...
    ...
    private final MeterIdPrefixFuntion meterIdPrefixFuntion;
    ...
    AuthService(HttpService delegate, Authorizer<HttpRequest> authorizer, MeterIdPrefixFunction meterIdPrefixFuntion, ...) {...}

    @Override
    public HttpResponse serve(ServiceRequestContext ctx, HttpRequest req) throws Exception {
        final long startNanos = System.nanoTime();
        return HttpResponse.of(AuthorizerUtil.authorizeAndSupplyHandlers(authorizer, ctx, req)
                                             .handleAsync((result, cause) -> {
            try {
                ...
            } finally {
                final MeterIdPrefix meterIdPrefix =
                        meterIdPrefixFunction.activeRequestPrefix(ctx.meterRegistry(), ctx.log().partial());
                MoreMeters.newTimer(ctx.meterRegistry(), meterIdPrefix.name(), meterIdPrefix.tags())
                          .record(System.nanoTime() - startNanos, TimeUnit.NANOSECONDS);
            }
        }, ctx.eventLoop()));
    }
    ...
}    
```
We might also want to add a tag that indicates success or failure if needed.

Reply:
Sorry about the late reply and thanks @minwoox .

Like @minwoox  mentioned, I think we can accept a `MeterIdPrefixFunction` from the constructor.
Additionally, I agree it's useful to also add tags indicating success/failure.

Having said this, if this feels like too much work I think it's fine to do this in a separate iteration